components/CheckDocumentation/connections/BlockingQuery.tsx

```diff
@@ -0,0 +1,145 @@
+      </p>
+    </>
+  );
+};
+
+const BlockingQueryGuidance: React.FunctionComponent<CheckGuidanceProps> = ({
+  urls: { backendsUrl, SettingLink },
+  issue,
+}) => {
+  const Link = useSmartAnchor();
+  const issueDetails = issue && JSON.parse(issue.detailsJson);
+  const showAutovacuum =
+    !issueDetails || issueDetails["has_autovacuum_to_prevent_wraparound"];
+  return (
+    <div>
+      <h4>Impact</h4>
+      <p>
+        Queries that hold contended locks for a significant period of time may
+        end up preventing other queries from making progress. This can even
+        block simple SELECT queries, and in some cases, it can affect overall
+        system performance or it can cause the system to run out of connections.
+      </p>
+      <h4>Common Causes</h4>
+      <ul>
+        {showAutovacuum && (
+          <li>
+            <h5>Autovacuum (to prevent wraparound)</h5>
+            <p>
+              In general, autovacuum won't block other queries. If a query tries
+              to acquire a lock that conflicts with a running autovacuum,
+              autovacuum will be interrupted so that the query can proceed.
+              However, if autovacuum is running to prevent transaction ID
+              wraparound, it will not yield like this and will block other
+              queries trying to acquire conflicting locks.
+            </p>
+          </li>
+        )}
+        <li>
+          <h5>Long-running migration</h5>
+          <p>
+            If some of these queries are DDL that is part of a database
+            migration, they may need to rewrite a table (e.g., changing the type
+            of a column) or scan a whole table (e.g., <code>CREATE INDEX</code>{" "}
+            without <code>CONCURRENTLY</code>), which can take much longer in
+            production than in a staging or development environment. If your
+            migration is transactional, consider canceling it and refactor it to
+            avoid table rewrites.
+          </p>
+        </li>
+        <li>
+          <h5>Other long-running queries</h5>
+          <p>
+            A long running query could cause many blocking queries when that
+            query is holding locks that conflict with many other queries.
+            Navigate to the <Link to={backendsUrl}>Connection Traces</Link> page
+            to check the overview of which query is blocking which queries to
+            understand the impact. It is useful to know what kind of locks the
+            blocking query is holding and what kind of locks the blocked queries
+            are waiting for. You can check the wait event of these locks in the{" "}
+            <strong>Wait Events</strong> page, with any wait events of type{" "}
+            <strong>Lock</strong>. If you have{" "}
+            <SettingLink setting="log_lock_waits" /> turned on, you can also see
+            details about locking in the <strong>Logs</strong> page for these
```

Comment:
Technically only on Scale, but I think it's okay to talk in passing about a feature users may not have.